.github/workflows/docker-push.yaml

```diff
@@ -19,76 +17,82 @@ jobs:
     permissions:
       contents: read
       packages: write
-      # This is used to complete the identity challenge
-      # with sigstore/fulcio when running outside of PRs.
       # id-token: write
 
     steps:
       - name: Check out the repo
         uses: actions/checkout@v4
 
-      # Install the cosign tool except for PR
-      # https://github.com/sigstore/cosign-installer
       # - name: Install cosign
       #   if: github.event_name != 'pull_request'
       #   uses: sigstore/cosign-installer@v3
 
-      # Set up BuildKit Docker container builder to be able to build
-      # multi-platform images and export cache
-      # https://github.com/docker/setup-buildx-action
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+
       - name: Set up Docker Buildx
         uses: docker/setup-buildx-action@v3
         with:
           platforms: linux/amd64, linux/arm64
 
-      # https://github.com/docker/login-action
       - name: Log in to GitHub Container Registry ${{ env.REGISTRY }}
-        if: github.event_name != 'pull_request'
         uses: docker/login-action@v3
         with:
           registry: ${{ env.REGISTRY }}
           username: ${{ github.actor }}
           password: ${{ secrets.GITHUB_TOKEN }}
 
-      # Extract metadata (tags, labels) for Docker
-      # https://github.com/docker/metadata-action
-      - name: Extract Docker metadata
-        id: meta
-        uses: docker/metadata-action@v5
-        with:
-          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+      # - name: Extract Docker metadata
+      #   id: meta
+      #   uses: docker/metadata-action@v5
+      #   with:
+      #     images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+
+      - name: Generate build ID
+        id: prep
+        run: |
+          branch="${GITHUB_REF##*/}"
+          timestamp=$(date +%Y%m%d-%H%M)
+          reverse_commit_hash=$(git rev-parse --short HEAD)
+          echo "BUILD_ID=${branch}-${timestamp}-${reverse_commit_hash}" >> $GITHUB_OUTPUT
 
       - name: Build and push Docker image
         uses: docker/build-push-action@v5
         with:
           context: .
-          push: ${{ github.event_name != 'pull_request' }}
-          tags: ${{ steps.meta.outputs.tags }}
-          labels: ${{ steps.meta.outputs.labels }}
+          push: false
+          tags: |
+            ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}:${{ steps.prep.outputs.BUILD_ID }}
+          file: ./Dockerfile
```

Comment:
Could be great to have a variable for this in case the Dockerfile is in another location